src/community/geofence-server/src/main/java/org/geoserver/geofence/web/GeofenceRulePage.java

```diff
@@ -0,0 +1,259 @@
+/* (c) 2015 Open Source Geospatial Foundation - all rights reserved
+ * (c) 2001 - 2015 OpenPlans
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.geofence.web;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.logging.Level;
+
+import org.apache.wicket.ajax.AjaxRequestTarget;
+import org.apache.wicket.ajax.form.OnChangeAjaxBehavior;
+import org.apache.wicket.markup.html.form.DropDownChoice;
+import org.apache.wicket.markup.html.form.Form;
+import org.apache.wicket.markup.html.form.IChoiceRenderer;
+import org.apache.wicket.markup.html.form.SubmitLink;
+import org.apache.wicket.markup.html.form.TextField;
+import org.apache.wicket.markup.html.link.BookmarkablePageLink;
+import org.apache.wicket.model.CompoundPropertyModel;
+import org.apache.wicket.model.ResourceModel;
+import org.geoserver.catalog.ResourceInfo;
+import org.geoserver.catalog.WorkspaceInfo;
+import org.geoserver.geofence.core.model.enums.GrantType;
+import org.geoserver.geofence.services.dto.ShortRule;
+import org.geoserver.platform.GeoServerExtensions;
+import org.geoserver.platform.Service;
+import org.geoserver.security.GeoServerRoleService;
+import org.geoserver.security.GeoServerSecurityManager;
+import org.geoserver.security.impl.GeoServerRole;
+import org.geoserver.security.impl.GeoServerUser;
+import org.geoserver.web.GeoServerApplication;
+import org.geoserver.web.GeoServerSecuredPage;
+import org.geoserver.web.wicket.ParamResourceModel;
+import org.springframework.dao.DuplicateKeyException;
+
+/**
+ * 
+ * Internal Geofence Rule Page
+ * 
+ * @author Niels Charlier
+ *
+ */
+@SuppressWarnings("serial")
```

Comment:
sure, copied that from other wicket pages. there are like 10 inline event classes that need one, but yeah why not